### PR TITLE
[CSL-969] Add helpers to work with `scrypt`

### PR DESCRIPTION
**Pull request by Martoon-00**
It was intended to change replace hashing of passphrase for secret key, but it will be done under CSL-1169 + some other issue, only helpers remained

### CODE REVIEW

**Comment by Martoon-00** [comment]
Requires hashing parameters tuning

File: core/Pos/Crypto/Scrypt.hs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import           Pos.Binary.Class  (Bi, encodeStrict)
+import           Pos.Crypto.Random (secureRandomBS)
+
+-- | This corresponds to 'ScryptParams' datatype.
+-- These parameters influence on resulting hash length, memory and time
+-- consumption. See documentation for exact details.
+data ScryptParamsBuilder = ScryptParamsBuilder
+    { spLogN    :: Word
+    , spR       :: Word
+    , spP       :: Word
```
Comment (rockbmb) [inline core/Pos/Crypto/Scrypt.hs]: Could you rename these 3 fields, or add some Haddock documentation to each so that their purpose becomes clearer?

Comment (Martoon-00) [inline core/Pos/Crypto/Scrypt.hs]: You're right, it became unclear. 
In your opinion, if I attach this link, will it be enough? https://hackage.haskell.org/package/scrypt-0.5.0/docs/Crypto-Scrypt.html

Comment (Martoon-00) [inline core/Pos/Crypto/Scrypt.hs]: Okay, I'll do my best to make it understandable in 2 minutes

**Comment by vincenthz** [comment]
another redundant package with cryptonite

**Comment by Martoon-00** [comment]
That `scrypt` library provides a more convenient way to go than the single function in `cryptonite`, but perhaps it doesn't worth having extra dependency, I don't even know whether will it be required later. So closing

**Pull request closed without merging**